overlord/hookstate/ctlcmd/health.go

```diff
@@ -0,0 +1,130 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/healthstate"
+)
+
+var (
+	shortHealthHelp = i18n.G("Report on snap's health")
+	longHealthHelp  = i18n.G(`
+The set-health command can signal to the system and the user that something is
+not well with the snap.
```

Comment:
Given that if there's a hook one must use this, the emphasis only on negative situations seems unwarranted. We need a more neutral explanation I think.

Also this should probably mention the hook as a common usage place.

Reply:
How about
```
The set-health command is called from a hook (primarily the "check-health" hook)
to inform the system of the overall health of the snap.
```
?

Reply:
It doesn't need to be called from hook, though the main usage is like that, no?  can be called from the snap at any point, or is that not true?

Reply:
You're right, a snap could call `snapctl set-health` from an app directly and maybe that's something we want to encourage, even. How about
```
The set-health command is called from a snap to inform the system of the snap's
overall health.
```
?

Reply:
actually that doesn't work, the only hook  that works is check-health, because how setting health in state is split out, but we want "snapctl set-health" to work at any point, so we need to think about that

Reply:
`snapctl set-health` now works from everywhere (hooks or apps). I'll be updating the help to the last one I suggested, in view of this. Shout if you want it different :-)